test/functional/rpc_fundrawtransaction.py

```diff
@@ -1200,6 +1201,67 @@ def test_add_inputs_default_value(self):
 
         self.nodes[2].unloadwallet("test_preset_inputs")
 
+    def test_preset_inputs_selection_crash(self):
+        self.log.info('Test wallet preset inputs selection')
+
+        # This exercises the bug presented in #26559 that
+        # ended up on (1) a wallet crash or (2) a created and
+        # broadcasted tx that pays an insane fee amount.
+        #
+        # This is covered by making the wallet selects the preset
+        # inputs twice during the coin selection process (which
+        # is a dangerous bug on v24.0).
+        #
+        # Making the wallet think that the selection process result covers
+        # the entire tx target when it does not. It's actually creating
+        # a tx that sends more coins than what inputs are covering for.
+        #
+        # Which, combined with the subtract fee from outputs option,
+        # makes the wallet generate and broadcast a tx that pays
+        # up to the sum of all the preset inputs, minus one, amounts in
+        # fee :(.
```

Comment:
If I were reading this a year from now (e.g. because I reintroduced the bug and this failed), I would need more help understanding what the test is doing. This comment is a bit too focused on the user effects of the bug (which is helpful but the commit message + PR description already says the same thing); the test should instead describe what unexpected code behavior it's aiming to catch.

```suggestion
        # Check that the wallet handles preset inputs properly,
        # i.e. rmoves them from the coin selection UTXO pool
        # and accurately accounts for the fee and value of all
        # inputs. Amounts should not be double-counted and
        # the wallet should not crash.
```